src/core/signal.h

```diff
@@ -60,4 +60,6 @@ class SignalManager : public Pimpl<SignalManager> {
     ~SignalManager();
 };
 
+extern SignalManager* signal_mgr;
```

Comment:
Why the switch to the global and manual memory management for it?

Reply:
The signal handler method for Windows is a global method and you can't pass any data into it directly (see `CtrlHandler` in signal.windows.cc. I need to be able to talk to the signal manager from that method, hence making it global. Along with this, the `sigint` variable created directly below `signal_mgr` needs to be destroyed before `signal_mgr` or it crashes, because of a dependency between the two. It's code that really should be reworked to not have that dependency. This was a quick fix and the signal management stuff can be cleaned up after this is merged.